src/Web/Streams/WriteableStream.purs

```diff
@@ -0,0 +1,37 @@
+module Web.Streams.WriteableStream
+  ( WriteableStream
+  , new
+  , abort
+  , close
+  , getWriter
+  , locked
+  ) where
+
+import Data.Maybe (Maybe)
+import Data.Nullable (Nullable, toNullable)
+import Effect (Effect)
+import Effect.Exception (Error)
+import Effect.Uncurried (EffectFn2, runEffectFn2)
+import Prelude (Unit)
+import Web.Promise (Promise)
+import Web.Streams.QueuingStrategy (QueuingStrategy)
+import Web.Streams.Sink (Sink)
+import Web.Streams.Writer (Writer)
+
+foreign import data WriteableStream :: Type -> Type
+
+foreign import _new :: forall chunk. EffectFn2 (Sink chunk) (Nullable (QueuingStrategy chunk)) (WriteableStream chunk)
+
+new :: forall chunk. Sink chunk -> Maybe (QueuingStrategy chunk) -> Effect (WriteableStream chunk)
+new source strategy = runEffectFn2 _new source (toNullable strategy)
+
+foreign import _abort :: forall chunk. EffectFn2 (WriteableStream chunk) Error (Promise Unit)
+
+abort :: forall chunk. WriteableStream chunk -> Error -> Effect (Promise Unit)
+abort = runEffectFn2 _abort
```

Comment:
MDN disagrees with the spec on type that should be passed to abort; [MDN claims that a string should be provided](https://developer.mozilla.org/en-US/docs/Web/API/WritableStream/abort), but the [spec is fine with any javascript value](https://streams.spec.whatwg.org/#writable-stream-abort). I went with error as that seems to be the pattern elsewhere, but I don't have a strong opinion on what should end up here.